test/should.js

```diff
@@ -1131,6 +1158,26 @@ describe('should', function() {
       testSet.should.not.have.any.keys([20, 1, {13: 37}]);
       testSet.should.not.have.all.keys([aKey, {'iDoNot': 'exist'}]);
 
+      // Using the same assertions as above but with `.deep` flag instead of using referential equality
```

Comment:
Same thing as mentioned in the assert interface tests about adding a non-deep test to demonstrate strict equality.